ci/scripts/build_example.sh

```diff
@@ -25,10 +25,27 @@ build_dir=${1}/build
 mkdir ${build_dir}
 pushd ${build_dir}
 
-cmake \
-    -DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX:-${ICEBERG_HOME}} \
-    ${source_dir}
-cmake --build .
+is_windows() {
+    [[ "${OSTYPE}" == "msys" || "${OSTYPE}" == "win32" ]]
+}
+
+CMAKE_ARGS=(
+    "-DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX:-${ICEBERG_HOME}}"
+)
+
+if is_windows; then
+    CMAKE_ARGS+=("-DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake")
+    CMAKE_ARGS+=("-DCMAKE_BUILD_TYPE=Release")
+else
+    CMAKE_ARGS+=("-DCMAKE_BUILD_TYPE=Debug")
+fi
```

Comment:
How about always using `-GNinja`?
If we use Ninja for Windows instead of `-G "Visual Studio 17 2022"`, we can use ccache. It will reduce CI time.